uaclient/tests/test_cli_status.py

```diff
@@ -818,6 +853,122 @@ def test_unicode_dash_replacement_when_unprintable(
         assert expected_out == out
 
+    @pytest.mark.parametrize(
+        "exception_to_throw,exception_type,exception_message",
+        (
+            (
+                exceptions.UrlError("Not found", 404),
+                exceptions.UrlError,
+                "Not found",
+            ),
+            (
+                exceptions.ContractAPIError(
+                    exceptions.UrlError("Unauthorized", 401),
+                    {"message": "unauthorized"},
+                ),
+                exceptions.UserFacingError,
+                "Invalid token. See https://ubuntu.com/advantage",
+            ),
+        ),
+    )
+    def test_errors_are_raised_appropriately(
+        self,
+        _m_getuid,
+        m_get_contract_information,
+        _m_get_avail_resources,
+        _m_should_reboot,
+        _m_remove_notice,
+        exception_to_throw,
+        exception_type,
+        exception_message,
+        capsys,
+        FakeConfig,
+    ):
+        """Check that simulated status json/yaml output raises errors."""
+
+        m_get_contract_information.side_effect = exception_to_throw
+
+        cfg = FakeConfig()
+
+        args = mock.MagicMock(
+            format="json", all=False, simulate_with_token="some_token"
+        )
+
+        with pytest.raises(Exception) as exc:
+            action_status(args, cfg=cfg)
+
+        assert exc.type == exception_type
+        assert exception_message in exc.value.args
+
+    @pytest.mark.parametrize(
+        "token_to_use, warning_message",
+        (
+            (
+                "expired_token",
+                'Contract "some_id" expired on December 31, 2019',
+            ),
+            (
+                "token_not_valid_yet",
+                'Contract "some_id" is not effective until December 31, 9999',
+            ),
+        ),
+    )
+    @pytest.mark.parametrize(
+        "format_type,event_logger_mode",
+        (("json", EventLoggerMode.JSON), ("yaml", EventLoggerMode.YAML)),
+    )
+    def test_warnings_for_token_dates(
+        self,
+        _m_getuid,
+        m_get_contract_information,
+        _m_get_avail_resources,
+        _m_should_reboot,
+        _m_remove_notice,
+        format_type,
+        event_logger_mode,
+        token_to_use,
+        warning_message,
+        capsys,
+        FakeConfig,
+        event,
+    ):
+        """Check that simulated status json/yaml output has warnings.
+
+        There must be warnings for expired tokens and for tokens not valid yet.
+        """
+
+        def contract_info_side_effect(cfg, token):
+            response = copy.deepcopy(RESPONSE_CONTRACT_INFO)
+            if token == "expired_token":
+                response["contractInfo"][
+                    "effectiveTo"
+                ] = "2019-12-31T00:00:00Z"
+            if token == "token_not_valid_yet":
+                response["contractInfo"][
+                    "effectiveFrom"
+                ] = "9999-12-31T00:00:00Z"
```

Comment:
I think we can set those variables using `parametrize`, like `contract_field` and `data_value`

Reply:
done